go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -54,7 +54,7 @@ func (bnc *BaseNetworkController) allocatePodIPsOnSwitch(pod *kapi.Pod,
 	// it is possible to try to add a pod here that has no node. For example if a pod was deleted with
 	// a finalizer, and then the node was removed. In this case the pod will still exist in a running state.
 	// Terminating pods should still have network connectivity for pre-stop hooks or termination grace period
-	if _, err := bnc.watchFactory.GetNode(pod.Spec.NodeName); kerrors.IsNotFound(err) &&
+	if node, err := bnc.watchFactory.GetNode(pod.Spec.NodeName); kerrors.IsNotFound(err) &&
```

Comment:
talked to @trozet here is recap
returning an error from `allocatePodIPsOnSwitch` during initial sync will cause restart. we know there is a k8s bug that can give us Running pod without existing node. Currently, for all non-terminated pods `waitForNodeLogicalSwitchSubnetsInCache` will wait for 30 seconds. Now we agreed that in case `waitForNodeLogicalSwitchSubnetsInCache` returns error for running pods without a node, we should ignore it considering that the node actually doesn't exist and won't come up similar to https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/base_network_controller_pods.go#L59-L62 and return error in other cases.
So you could set some bool like `podWithoutNode` under https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/base_network_controller_pods.go#L57-L58 and then conditionally return error from `waitForNodeLogicalSwitchSubnetsInCache` based on this value
hope this makes sense :) if not, please reach out to me

Reply:
Maybe a more elegant way to fix it is to make the error be a specific instance of a new type, lets call it WaitForNodeSwitchError. Then in syncPods, when this code happens:

```
			expectedLogicalPortName, annotations, err = oc.allocateSyncPodsIPs(pod)
			if err != nil {
				return err
			}
```

We check here if the error is WatiForNodeError and then just continue? I guess continue may not be the best option by itself, since we will continue to try to add every pod and wait 30 seconds for the switch, that we already waited 30 seconds for and it never showed up. So might need to store notFoundSwitches map[string]bool var or something and compare each pod so you can short-circuit pods that we already know arent going find the switch.